### ISSUE TITLE
Always display process name (if possible)

### ISSUE THREAD

**Issue by JayBrown** [theevilbit/Shield#5]
I know that whitelists are on the ToDo list, but for starters, a display of the process name in the Shield notification would be nice. It works for some injections, but not all.

Example: I'm currently getting injection warnings during normal browser operations.

Safari extensions produce a dylib hijacking warning, and the path is included.

In the Firefox browser, however, I'm getting an environment variable injection warning, but no path, just "victim process", so I can't say what process is behind it. It happens whenever I open a URL/link in a new browser tab, so I assume that it's the system-wide Adguard adblocker (running as a network extension)… but it might also be 1Password or the Adobe PDF web extension (NativeMessagingHost).

Or it's Firefox itself and its extensions. (?)

EDIT: it's not Adguard, because other apps/browsers also managed by Adguard do not spawn a Shield notification.

**Comment by theevilbit** [comment]
I need to work on a much better alerting, it's on my TODO list and include many info, like the one you suggested.
This is a Firefox issue. You can also check the logs at `/Library/Application\ Support/Shield/shield.log`.
```
2021-01-13 11:54:35 +0000: Environment variable (DYLD_INSERT_LIBRARIES) injection blocked
Victim process:
/Applications/Firefox.app/Contents/MacOS/plugin-container.app/Contents/MacOS/plugin-container
```

Some apps launch some internal apps with `DYLD_INSERT_LIBRARIES`. It's also true for GitHub Desktop for example.

Any 3rd party on any app like your Safari case will produce dylib hijack alerts, as in the background I enforce library validation, only dylibs signed by the same developer are allowed to load. Also a future TODO :)

**Comment by theevilbit** [comment]
Better alerting is implemented with https://github.com/theevilbit/Shield/commit/4fc0af77b90d021904067d0823d67fd4d88d0b0b

**Comment by JayBrown** [comment]
Shield is performing a lot better now, not in Firefox yet, but in general: all the other false positives are gone, which in my case were benign background processes of programs and CLIs, e.g. iTerm launch, wget, Homebrew, GitHub Desktop + git, F-Secure fsav anti-malware updates etc. No injection warnings in this respect any more.

**Comment by theevilbit** [comment]
Is it running properly? GitHub Desktop does raise an alert.

**Comment by theevilbit** [comment]
I haven't changed anything in the logic, so it should alert for the exact same processes.

**Comment by JayBrown** [comment]
That's strange. Shield seems to be running fine (see Firefox below), but I'm not getting the other alerts anymore (Homebrew, GitHub Desktop, F-Secure etc.)

![snap](https://i.imgur.com/jCA9QsW.jpg)

**Comment by JayBrown** [comment]
I uninstalled the system extension, quit Shield, relaunched it, and then macOS told me that the extension was blocked, so I approved it again… still not getting alerts except in Firefox (env vars).

**Comment by theevilbit** [comment]
All injections are set to blocking? I think brew alerts with dylib hijacking. Github only alerts when you push/pull, just running it doesn’t trigger. How does your preferences look?

**Comment by JayBrown** [comment]
```
{
    isBlocking = 0;
    isRunning = 1;
    prefDylib = 1;
    prefElectron = 1;
    prefEnvVars = 0;
    prefTFP = 1;
    skipApple = 1;
}
```

So I've currently disabled alerts for EnvVars, but the same behavior occurs when I also disable it.

**Comment by theevilbit** [comment]
Just tested, and GitHub Desktop does alert when doing a push. It's an ENVVAR type of injection.
<img width="728" alt="Screen Shot 2021-02-09 at 11 15 35" src="https://user-images.githubusercontent.com/10151085/107349315-5c54d900-6ac8-11eb-80c6-127f45656bf0.png">

**Comment by theevilbit** [comment]
OK, I see why there is no alerts for dylib stuff. I put a line of code in the wrong place, you should still get alerts of in "detection" mode.

**Comment by theevilbit** [comment]
Released version 0.9.5.1. How does it look?
https://github.com/theevilbit/Shield/commit/13c48e01fc9cf1ccb4f062c2dfbf000ef648239d

**Comment by theevilbit** [comment]
Ok, fixed dylib injection issue. I show everything I can now with https://github.com/theevilbit/Shield/commit/984354d822da31e21d1df9e7f5408a716703e8c5.

If you need to see more details please open another Issue. Thanks!
